python/lsst/ip/isr/ampOffset.py

```diff
@@ -0,0 +1,86 @@
+# This file is part of ip_isr.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+# import os
+
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+from lsst.meas.algorithms import (SubtractBackgroundTask, SourceDetectionTask)
+
+
+class AmpOffsetConfig(pexConfig.Config):
+    """Configuration parameters for AmpOffsetTask.
+    """
+    ampEdgeInset = pexConfig.Field(
+        doc="Number of pixels the amp edge strip is inset from the amp edge. A thin strip of pixels running "
+        "parallel to the edge of the amp is used to characterize the average flux level at the amp edge.",
+        dtype=int,
+        default=5,
+    )
+    ampEdgeWidth = pexConfig.Field(
+        doc="Pixel width of the amp edge strip, starting at ampEdgeInset and extending inwards.",
+        dtype=int,
+        default=64,
+    )
+    ampEdgeMinFrac = pexConfig.Field(
+        doc="Minimum allowed fraction of viable pixel rows along an amp edge. No amp-to-amp step estimate "
+        "will be generated for amp edges that do not have at least this fraction of unmasked pixel rows.",
+        dtype=float,
+        default=0.5,
+    )
+    ampEdgeMaxStep = pexConfig.Field(
+        doc="Maximum allowed amp-to-amp step value. If a measured amp-to-amp step value is larger than this, "
+        "the result will be discarded and therefore not used to determine amp offset piston corrections.",
```

Comment:
We have (starting here in my review) the use of `amp-to-amp step` and `amp offset piston`.  I think we need to (a) standardize on a name, and (b) the more I think about it I don't like `piston` because it confused Josh (there's a physical up-down piston which relates to focus).  I'm not sure what name to use, maybe we should open this up on #dm-naming-things.

Reply:
The doc string should specify the units (ADU).

Reply:
Thanks for these comments Eli, and apologies it's taken me some time to reply. So, following your comments here and the previous discussion on #dm-naming-things, I have now standardized on two names to use for this ticket: amp offset describes the jump (in ADU) from one amp to another, and amp pedestal describes the specific value (in ADU) added/subtracted from an individual amplifier. All mentions of 'step' and 'piston' have been removed from the code and associated doc strings. I hope that reads well, but keen to hear your thoughts if not.

Reply:
PS - mention of units (ADU) added to the docstring here also, thank you.